api/envoy/service/status/v3/csds.proto

```diff
@@ -53,6 +52,24 @@ enum ConfigStatus {
   ERROR = 4;
 }
 
+// Config status from a client-side view.
+enum ClientConfigStatus {
+  // Config status is not available/unknown.
+  CLIENT_UNKNOWN = 0;
+
+  // Client requested the config but hasn't received any config from management
+  // server yet.
+  CLIENT_REQUESTED = 1;
+
+  // Client received the config and replied with ACK.
+  CLIENT_ACKED = 2;
+
+  // Client received the config and replied with NACK. Notably, the attached
+  // config dump is not the NACKed version, but the most recent accepted one. If
+  // no config is accepted yet, the attached config dump will be empty.
```

Comment:
Can we have a similar statement added about control plane behavior for configs with a negative status, i.e. "this is always the latest successful config?" Or maybe that's not even desired, we might want the latest successful and unsuccessful config? Would that also make sense for client? CC @fuqianggao

Reply:
This behavior wasn't clearly described in this proto definition before, or in other documentation. Could it be a control-plane specific detail?

Hi @fuqianggao, can you help to explain the designed behavior of config dump when client replied with NACK?

Reply:
For control plane, it will be the latest config that control plane has. If status is NACK, it means the config is sent to client, but rejected.

Reply:
Makes sense. I think the actionable thing is for @lidizheng to update the management server wording to reflect this. I'm still wondering in this discussion thread, is there value in having both active and failed configs reported?

Reply:
I suspect that for both xDS servers and xDS clients, there would be additional overhead in storing old versions of configs.  In general, both sides will store only the latest version that they care about, so they don't have anything else to report here.

Reply:
And if all clients implement CSDS, the CLI can get the active config from the clients, and the most up-to-date config from the management server (if they are different).

Reply:
I have updated the comment for `ConfigStatus.ERROR` to note that the management server will dump the latest config in case of client replying NACK.

Reply:
I like the idea of being able to query both client and server CSDS and effectively determine the failing and active config. @lidizheng can you please file an issue for Envoy to add support for CSDS? You don't have to do the implementation, but we should track this as an open feature request that we can share with the community.

Reply:
Filed as an issue: https://github.com/envoyproxy/envoy/issues/13181. For Envoy, it overlaps with the admin config dump feature. Today, tools like `istioctl` is fetching xDS config from server via CSDS, and from client via admin config dump (HTTP).